XYZDrivers/src/main/java/com/xyzdrivers/controllers/ClaimStatusController.java

```diff
@@ -0,0 +1,113 @@
+package com.xyzdrivers.controllers;
+
+import com.xyzdrivers.models.Claim;
+import com.xyzdrivers.repositories.ClaimsRepo;
+import com.xyzdrivers.repositories.RepositoryException;
+import com.xyzdrivers.services.ResponseService;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ *
+ * @author arthur
+ */
+public class ClaimStatusController extends HttpServlet {
+    
+    @Inject
+    ClaimsRepo claimsRepo;
+
+    /**
+     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
+     * methods.
+     *
+     * @param request servlet request
+     * @param response servlet response
+     * @throws ServletException if a servlet-specific error occurs
+     * @throws IOException if an I/O error occurs
+     */
+    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        
+        // Get the http request params
+        String status = request.getParameter("status");
+        Integer id = Integer.parseInt(request.getParameter("claim_id"));
+        
+        // Sanity check to ensure required parameters were required, if not respond with failure
+        if (status == null || status.equals("")) {
+            ResponseService.fail(request, response, "Failure. Please submit a status and claim id", "admin");
+            return;
+        }
+        
+        // Get the claim to be updated by id
+        Claim claim = claimsRepo.get(id);
+        
+        // Check we found the claim by ID
+        if (claim == null) {
+            ResponseService.fail(request, response, "Failure. The claim could not be found by id", "admin");
+            return;
+        }
+        
+        // Set the status using model's setter to ensure status is valid, if not respond with failure
+        if (! claim.setStatus(status)) {
+            ResponseService.fail(request, response, "Failure. Please submit a valid status", "admin");
+            return;
+        }
+        
+        try {
+            // Update the model and persist to DB
+            claim = claimsRepo.update(claim);
+        } catch (RepositoryException ex) {
+            Logger.getLogger(ClaimStatusController.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
If there was an exception updating we should not be telling the user we succeeded